rust/lance/src/index/vector/ivf.rs

```diff
@@ -2267,24 +2267,58 @@ mod tests {
         .await
         .unwrap();
 
-        let index = dataset
```

Comment:
are these changes related? Cannot connect why we need these changes

Reply:
These changes are introduced due to the dependency of open_index like this:

```
        let remapped = dataset_mut
            .open_vector_index(
                WellKnownIvfPqData::COLUMN,
                &new_uuid.to_string(),
                &NoOpMetricsCollector,
            )
            .await
            .unwrap();
```

The testcase would open index after building, now open_index demands index has been registered in dataset(The testcase worked well because open_index only needs uuid even though the index can not been seen in the dataset).